scene/gui/tab_bar.cpp

```diff
@@ -1273,11 +1273,9 @@ Variant TabBar::_handle_get_drag_data(const String &p_type, const Point2 &p_poin
 
 bool TabBar::_handle_can_drop_data(const String &p_type, const Point2 &p_point, const Variant &p_data) const {
```

Comment:
This does not get called when `drag_to_rearrange_enabled` is not enabled, so the `can_drop_data` for this and TabContainer should be changed to also check `switch_on_hover`.